src/Integration.UnitTests/MefServices/ActiveSolutionBoundTrackerTests.cs

```diff
@@ -261,24 +252,188 @@ public void ActiveSolutionBoundTracker_Changes()
             this.errorListController.ResetCalledCount.Should().Be(0);
             solutionBindingChangedEventCount.Should().Be(5, "Solution change should trigger reanalysis");
 
-            // We don't call SonarQubeService.Disconnect directly anymore, we execute the DisconnectCommand, which calls it
-            mockSqService.Verify(disconnectMethod, Times.Never);
-            mockSqService.Verify(connectMethod, Times.Exactly(1));
+            // SonarQubeService.Disconnect should be called since the WPF DisconnectCommand is not available
+            VerifyServiceDisconnect(Times.Once());
+            VerifyServiceConnect(Times.Once());
 
             // Case 6: Dispose and change
             // Act
             testSubject.Dispose();
-            solutionBinding.CurrentBinding = null;
-            this.solutionBindingInformationProvider.SolutionBound = true;
+            SetSolutionBinding(boundProject);
             host.VisualStateManager.ClearBoundProject();
 
             // Assert
             solutionBindingChangedEventCount.Should().Be(5, "Once disposed should stop raising the event");
             // TODO: this.errorListController.RefreshCalledCount.Should().Be(3);
             this.errorListController.ResetCalledCount.Should().Be(1);
-            // We don't call SonarQubeService.Disconnect directly anymore, we execute the DisconnectCommand, which calls it
-            mockSqService.Verify(disconnectMethod, Times.Never);
-            mockSqService.Verify(connectMethod, Times.Exactly(1));
+            // SonarQubeService.Disconnect should be called since the WPF DisconnectCommand is not available
+            VerifyServiceDisconnect(Times.Once());
+            VerifyServiceConnect(Times.Once());
+        }
+
+        [TestMethod]
+        public void UpdateConnection_WasDisconnected_NewSolutionIsUnbound_NoConnectOrDisconnectCalls()
+        {
+            // Arrange
+            var testSubject = new ActiveSolutionBoundTracker(this.host, this.activeSolutionTracker, sonarLintOutputMock.Object);
+
+            // Act
+            SetupService(isConnected: false);
+            SetSolutionBinding(null);
```

Comment:
Strictly these two _setup_ calls are part of _Arrange_ rather than _Act_. I'll move them.